### ISSUE TITLE
Implement Trueskill 2.0 features

### ISSUE THREAD

**Issue by delaneyj** [mafredri/go-trueskill#5]
[The Trueskill 2.0 paper came out a few months ago.](https://www.microsoft.com/en-us/research/uploads/prod/2018/03/trueskill2.pdf)  It appears to be taking in game per players stats into account and includes quit penalties.  There claim is much higher predictive power.  It however doesn't appear to come with any source code.  Is it possible to infer that steps necessary to implement?

**Comment by mafredri** [comment]
Thanks for bringing it to my attention. I’m going to put it on my reading list. 

We could probably implement if they havent omitted too much info / details from the paper.

**Comment by delaneyj** [comment]
Yeah, I actually ended up [porting the python version of the original paper](https://github.com/delaneyj/trueskill) over just to begin to understand the new paper.  My version supports multiple teams and doesn't use the bag collection that yours does.  Thought we might be able to decode the new paper together.
